### ISSUE TITLE
clearTimeout with null/undefined throws exception

### ISSUE THREAD

**Issue by teropa** [angular/zone.js#276]
When using `clearTimeout` with an `undefined` or `null` value on a page with Zone.js, the following exception is thrown:

```
zone.js:115 Uncaught TypeError: Cannot read property 'zone' of undefined
```

Plunker: https://plnkr.co/edit/dDEXwEVm3hbmleKU179b?p=preview

This should not happen, as [MDN says](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/clearTimeout): "Passing an invalid ID to clearTimeout does not have any effect (and doesn't throw an exception)."

**Comment by mhevery** [comment]
Closed by #279

**Comment by justindujardin** [comment]
@mhevery are you sure this is fixed by #279 ? I'm seeing a second behavior here which is an opinionated exception when using timeouts.

If I clear a timeout twice I will get an exception: `Uncaught Error: More tasks executed then were scheduled.`.  

```typescript
var timeout = setTimeout(() => {}, 100);
clearTimeout(timeout);
clearTimeout(timeout);
```

I've been adding defensive guards around this behavior in my various apps, but I think it's pretty common for users to call clearTimeout on invalid handles in normal use-cases.  The browser won't complain, so should zone.js?

**Comment by mhevery** [comment]
@justindujardin yes this should be fixed in the latest zone. https://github.com/angular/zone.js/blob/master/test/browser/setTimeout.spec.ts#L55
